File: app/src/main/java/com/sp/base/ui/DatabaseOprationActivity.java
package com.sp.base.ui;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.util.Log;

import com.sp.base.R;
import com.sp.base.appbase.BaseApplication;
import com.sp.base.database.AppDatabase;
import com.sp.base.database.UserProfileDetailsModel;

public class DatabaseOprationActivity extends AppCompatActivity {
    private AppDatabase mDb;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        DataBindingUtil.setContentView(this, R.layout.activity_database_opration);
        mDb = BaseApplication.getInstance().getAppDatabase();

    }

    public void getDataFromDB() {
        UserProfileDetailsModel userProfileDetailsModel = mDb.getDbDAO().getUserProfileDetails();
        Log.e("db", "getDataFromDB: " + userProfileDetailsModel);
    }

    public void setDataFromDB() {
        UserProfileDetailsModel userProfileDetailsModel = new UserProfileDetailsModel();

        mDb.getDbDAO().insertUserData(userProfileDetailsModel);
        Log.e("db", "getDataFromDB: " + userProfileDetailsModel);
    }
}

File: app/src/main/java/com/baseapplication/ui/DatabaseOprationActivity.kt
package com.baseapplication.ui

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.baseapplication.BaseApplication
import com.baseapplication.R
import com.baseapplication.database.AppDatabase
import com.baseapplication.database.UserProfileDetailsModel

//@AndroidEntryPoint
class DatabaseOprationActivity() : AppCompatActivity() {
    private var mDb: AppDatabase? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DataBindingUtil.setContentView<ViewDataBinding>(this, R.layout.activity_database_opration)
        mDb = BaseApplication.instance?.appDatabase
    }

    val dataFromDB: Unit
        get() {
            val userProfileDetailsModel: UserProfileDetailsModel? =
                mDb?.dbDAO?.userProfileDetails
            Log.e("db", "getDataFromDB: " + userProfileDetailsModel)
        }

    fun setDataFromDB() {
        val userProfileDetailsModel: UserProfileDetailsModel = UserProfileDetailsModel()
        mDb?.dbDAO?.insertUserData(userProfileDetailsModel)
        Log.e("db", "getDataFromDB: " + userProfileDetailsModel)
    }
}